src/app/console.py

```diff
@@ -1,58 +1,188 @@
+        print('---------')
+        print(f'\033[1m\033[33mPreostali koraki {self.maxKoraki} \033[0m')
+        print('-------------------')
+        for y in range(self.zemlja.visina + 1):
+            for x in range(self.zemlja.sirina + 1):
                 flag = False
                 for b in self.zemlja.barabe:
                     if b.x == x and b.y == y:
-                        print('B ',end='')
+                        print('\033[91mB\033[0m ', end='')
                         flag = True
                         break
                 if flag:
                     continue
                 if x == self.zemlja.hero.x and y == self.zemlja.hero.y:
-                    print('H ', end='')
+                    print('\033[92mH\033[0m ', end='')
+                elif x == self.zemlja.princeska.x and y == self.zemlja.princeska.y:
+                    print('\u2665 ', end='')
                 else:
-                   print('. ', end='')
+                    print('. ', end='')
             print()
 
-
+    # premik heroja z inputi
     def input(self):
-        dx = int(input("Vnesi dx: "))
-        dy = int(input("Vnesi dy: "))
-        self.zemlja.hero.premikanje(dx=dx, dy=dy)
+        global dx
+        global dy
+        self.maxKoraki -= 1
+
+        x, y = self.zemlja.hero.x, self.zemlja.hero.y
+
+        while True:
+            uporabniskiInput = input('\033[92mVnesi dx med +2 in -2:\033[0m ')
+            if uporabniskiInput == "q":
+                sys.exit()
+            try:
+                dx = int(uporabniskiInput)
+                if not (-2 <= dx <= 2):
+                    print('\033[31mVnesli se številko izven razpona, ponovno vnesite številko dx.\033[0m')
+                else:
+                    break
+            except ValueError:
+                print('\033[31mProsim vnesite številko dx med +2 in -2.\033[0m')
+        while True:
+            uporabniskiInput = input('\033[92mVnesi dy med +2 in -2:\033[0m ')
+            if uporabniskiInput == "q":
+                sys.exit()
+            try:
+                dy = int(uporabniskiInput)
+                if not (-2 <= dy <= 2):
+                    print('\033[31mVnesli se številko izven razpona, ponovno vnesite številko dy.\033[0m')
+                else:
+                    break
+            except ValueError:
+                print('\033[31mProsim vnesite številko dy med +2 in -2.\033[0m')
+
+        # če si izven omejitve zemlje se heroj izriše na drugi strani
+        if x + dx > self.zemlja.sirina:
+            self.zemlja.hero.x = x + dx - self.zemlja.sirina - 1
+        elif x + dx < 0:
+            self.zemlja.hero.x = x + dx + self.zemlja.sirina + 1
+        elif y + dy > self.zemlja.visina:
+            self.zemlja.hero.y = y + dy - self.zemlja.visina - 1
+        elif y + dy < 0:
+            self.zemlja.hero.y = y + dy + self.zemlja.visina + 1
+        else:
+            self.zemlja.hero.premikanje(dx=dx, dy=dy)
+
+    def premakni_barabo(self):
```

Comment:
Ta metoda ne spada v consolo pač pa v class zemlja...
V primeru če imaš akcijo v kateri se ustvari sprememba več objektov različnih tipov, ponavadi daš metodo akcije na starševski objekt ki vsebuje vse te objekte, v najinem primeru je prvi objekt ki vsebuje vse barabe zemlja in ne konsola.